core/trino-main/src/main/java/io/trino/execution/CommentTask.java

```diff
@@ -86,20 +88,37 @@ else if (statement.getType() == Comment.Type.COLUMN) {
             }
 
             QualifiedObjectName tableName = createQualifiedObjectName(session, statement, prefix.get());
-            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
-            if (tableHandle.isEmpty()) {
-                throw semanticException(TABLE_NOT_FOUND, statement, "Table does not exist: " + tableName);
-            }
-
             String columnName = statement.getName().getSuffix();
-            Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle.get());
-            if (!columnHandles.containsKey(columnName)) {
-                throw semanticException(COLUMN_NOT_FOUND, statement, "Column does not exist: " + columnName);
+            Optional<ViewDefinition> optionalView = metadata.getView(session, tableName);
+            if (optionalView.isPresent()) {
+                ViewDefinition viewDefinition = optionalView.get();
+                List<ViewColumn> viewColumns = viewDefinition.getColumns();
+                Optional<ViewColumn> optionalViewColumn = viewColumns
+                        .stream()
+                        .filter(viewColumn -> viewColumn.getName().equals(columnName))
+                        .findFirst();
+
+                if (optionalViewColumn.isEmpty()) {
+                    throw semanticException(COLUMN_NOT_FOUND, statement, "Column does not exist: " + columnName);
+                }
+
+                accessControl.checkCanSetColumnComment(session.toSecurityContext(), tableName);
```

Comment:
>  was able to add comment to viewdefiniton

The comment isn't yet stored to HMS as far as I checked. How did you confirm the comment?